data/pools/transactionPool.go

```diff
@@ -220,17 +245,42 @@ func (pool *TransactionPool) rememberCommit(flush bool) {
 	if flush {
 		pool.pendingTxGroups = pool.rememberedTxGroups
 		pool.pendingTxids = pool.rememberedTxids
+		pool.pendingLastestLocal = pool.rememberedLatestLocal
 		pool.ledger.VerifiedTransactionCache().UpdatePinned(pool.pendingTxids)
 	} else {
+		// update the GroupCounter on all the transaction groups we're going to add.
+		// this would ensure that each transaction group has a unique monotonic GroupCounter
+		encodingBuf := protocol.GetEncodingBuf()
+		for i, txGroup := range pool.rememberedTxGroups {
+			pool.pendingCounter++
+			txGroup.GroupCounter = pool.pendingCounter
+			txGroup.EncodedLength = 0
```

Comment:
This is recalculating `txGroup.EncodedLength`. Wasn't it correct before?

Reply:
it's not that it wasn't correct before - it was never populated there before. The first time where we calculate this value is when a transaction group is being added to the transaction pool - which is exactly here.